test/Parser.test.js

```diff
@@ -227,9 +241,9 @@ describe("Parser", function() {
 			"typeof 'str'": "string=string",
 			"typeof aString": "string=string",
 			"typeof b.Number": "string=number",
-			"typeof b[Number]": "",
```

Comment:
And they will propably fail, which is because your code isn't handling that case correctly.